libs/designsystem/dropdown/src/dropdown.component.ts

```
@@ -525,6 +524,104 @@ export class DropdownComponent implements AfterViewInit, OnDestroy, ControlValue
     }
   }
 
+  private preventDefaultAndStopImmediatePropagation(event: KeyboardEvent) {
+    event.stopImmediatePropagation();
+    event.preventDefault();
+  }
+
+  private searchBuffer = '';
+  private searchBufferTimeout: any;
+  private SEARCH_BUFFER_DELAY = 500; // ms
+
+  private addToSearchBuffer(char: string) {
+    clearTimeout(this.searchBufferTimeout);
+    this.searchBuffer += char;
+    this.searchBufferTimeout = setTimeout(() => {
+      this.resetSearchBuffer();
+    }, this.SEARCH_BUFFER_DELAY);
+  }
+  private resetSearchBuffer() {
+    this.searchBuffer = '';
+  }
+
+  @HostListener('keydown', ['$event'])
+  _onKeydown(event: KeyboardEvent): boolean {
+    const key = event.key;
+
+    if (this.items.length === 0) {
+      console.warn('[Kirby] No items found within dropdown');
+      return false;
+    }
+
+    if (this.disabled) return false;
+
+    this.handlePrintableCharacterKey(event, this.isOpen);
+
+    // ALT + ArrowDown: Open dropdown
+    if (key === 'ArrowDown' && event.altKey) {
+      if (!this.isOpen) {
+        event.preventDefault();
+        this.open();
+        this.focusedIndex = this.selectedIndex > -1 ? this.selectedIndex : -1;
+      }
+      return false;
+    }
+
+    // ALT + ArrowUp: Select focused item and close dropdown
+    if (key === 'ArrowUp' && event.altKey) {
+      this.preventDefaultAndStopImmediatePropagation(event);
+      if (this.focusedIndex > -1) {
+        this.selectItem(this.focusedIndex);
+        this.close();
+      }
+      return false;
+    }
+    return true;
+  }
+
+  private handlePrintableCharacterKey(event: KeyboardEvent, isOpen: boolean) {
+    const key = event.key;
+
+    if (!SearchHelper.isPrintableCharacter(key)) {
+      return;
+    }
+    if (isOpen) {
+      this.preventDefaultAndStopImmediatePropagation(event);
```

Why only stop propagation if this is open?

---

Open: Block default and bubbling to ensure dropdown keyboard behavior is correct and isolated.
Closed: Let the event propagate, since the dropdown is not yet active